indico/modules/attachments/api/util.py

```diff
@@ -111,7 +111,8 @@ def _build_attachment_api_data(attachment):
         'description': attachment.description,
         'modified_dt': attachment.modified_dt.isoformat(),
         'type': attachment.type.name,
-        'is_protected': attachment.is_protected
+        'is_protected': attachment.is_protected,
+        'checksum': attachment.file.md5,
```

Comment:
This will break for link attachments. Add it in the if type == file block below.

Also, usually you should not use your own `master` branch for PRs - always create a different branch. We often rebase+merge or force-push small changes (e.g. changing the wording in changelog entries) and it's a bit of a mess.

In fact this is a PR I will most likely rebase to 3.1.x, which means I would need to force-push 3.1.x to your master, leaving your master in a weird state afterwards - with a throwaway feature branch that's no problem since you'd delete that after the PR is merged.